source/MaterialXRuntime/RtNodeImpl.cpp

```diff
@@ -0,0 +1,71 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/RtNodeImpl.h>
+#include <MaterialXRuntime/Tokens.h>
+
+#include <MaterialXRuntime/Private/PvtPath.h>
+#include <MaterialXRuntime/Private/PvtPrim.h>
+
+namespace MaterialX
+{
+
+DEFINE_TYPED_SCHEMA(RtNodeImpl, "nodeimpl");
+
+RtPrim RtNodeImpl::createPrim(const RtToken& typeName, const RtToken& name, RtPrim parent)
+{
+    if (typeName != _typeInfo.getShortTypeName())
```

Comment:
Just a thought, but you seem to do these a lot, is it worth having a shared function call for these checks? You can pass in a string for the "nodeimpl" part or have a typeName() function which returns "nodeimpl" that each derived class could implement. Just a thought.